test/jdk/java/awt/regtesthelpers/UtilInternal.java

```diff
@@ -73,6 +73,12 @@ public static Frame createEmbeddedFrame(final Frame embedder)
             Class clazz = Class.forName("sun.awt.X11.XEmbeddedFrame");
             Constructor constructor = clazz.getConstructor (new Class [] {Long.TYPE, Boolean.TYPE});
             return (Frame) constructor.newInstance (new Object[] {window, true});
+        } else if ("sun.lwawt.macosx.LWCToolkit".equals(tk.getClass().getName())) {
+            java.awt.Helper.addExports("sun.lwawt", UtilInternal.class.getModule());
+            Class window_peer_class = Class.forName("sun.lwawt.LWComponentPeer");
+            Method get_target = window_peer_class.getMethod("getTarget", new Class[0]);
+            System.out.println("get_target = " + get_target);
+            return (Frame) get_target.invoke(frame_peer, new Object[0]);
```

Comment:
Shouldn't we invoke sun/lwawt/macosx/CEmbeddedFrame same as we do for windows/linux via WEmbeddedFrame and XEmbeddedFrame?

Reply:
CEmbeddedFrame doesn't have a way to pass in a handle like in WEmbeddedFrame and XEmbeddedFrame, so I can't pass in the peer's target. If I try to use CEmbeddedFrame the test window will pop up but the EmbeddedFrame won't show up and the test fails

Reply:
Are you sure that CEmbeddedFrame is used in the embedding functionality? We have another one on macOS CViewEmbeddedFrame. One of them was used for applets another one was used by the JAWT and SWT. Please double-check that the correct one is used for this test, and another one probably can be removed.

Reply:
It seems LWCToolkit creates a CViewPlatformEmbeddedFrame for VIEW_EMBEDDED_FRAME type so probably you can call CViewPlatformEmbeddedFrame.getNSViewPtr to get the long value and then call CViewEmbeddedFrame with the nsviewPtr value